tools/export/codeblocks/__init__.py

```diff
@@ -0,0 +1,152 @@
+"""
+mbed SDK
+Copyright (c) 2014-2017 ARM Limited
+Copyright (c) 2018 Code::Blocks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import copy
+import os
+from os.path import splitext, basename
+from os import remove
+from shutil import rmtree
+from tools.targets import TARGET_MAP
+from tools.export.exporters import Exporter
+from tools.export.makefile import GccArm
+
+class CodeBlocks(GccArm):
+    NAME = 'Code::Blocks'
+
+    DOT_IN_RELATIVE_PATH = True
+
+    MBED_CONFIG_HEADER_SUPPORTED = True
+
+    PREPROCESS_ASM = False
+
+    POST_BINARY_WHITELIST = set([
+        "NCS36510TargetCode.ncs36510_addfib"
+    ])
+
+    @staticmethod
+    def filter_dot(str_in):
+        """
+        Remove the './' prefix, if present.
+        This function assumes that resources.win_to_unix()
+        replaced all windows backslashes with slashes.
+        """
+        if str_in is None:
+            return None
+        if str_in[:2] == './':
+            return str_in[2:]
+        return str_in
+
+    def generate(self):
+        self.resources.win_to_unix()
+
+        comp_flags = []
+        debug_flags = []
+        next_is_include = False
+        for f in self.flags['c_flags'] + self.flags['cxx_flags'] + self.flags['common_flags']:
+            f=f.strip()
+            if f == "-include":
+                next_is_include = True
+                continue
+            if f == 'c':
+                continue
+            if next_is_include:
+                f = '-include ' + f
+            next_is_include = False
+            if f.startswith('-O') or f.startswith('-g'):
+                debug_flags.append(f)
+                continue
+            comp_flags.append(f)
+        comp_flags = list(set(comp_flags))
+        inc_dirs = [self.filter_dot(s) for s in self.resources.inc_dirs];
+        inc_dirs = [x for x in inc_dirs if x is not None and x != '' and x != '.' and not x.startswith('bin') and not x.startswith('obj')];
+
+        c_sources = [self.filter_dot(s) for s in self.resources.c_sources]
+        targ = TARGET_MAP[self.target]
```

Comment:
You will not have any configuration variables overridden on this target, making configuration incorrect. Please use `self.toolchain.target` instead.